tests/cppunit/disk_test.cc

```diff
@@ -0,0 +1,202 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include <chrono>
+#include <memory>
+#include <string>
+#include <vector>
+#include <gtest/gtest.h>
+#include "redis_metadata.h"
+#include "test_base.h"
+#include "redis_string.h"
+#include "redis_disk.h"
+#include "redis_set.h"
+#include "redis_list.h"
+#include "redis_zset.h"
+#include "redis_bitmap.h"
+#include "redis_sortedint.h"
+#include "redis_stream.h"
+
+class RedisDiskTest : public TestBase {
+protected:
+  explicit RedisDiskTest() : TestBase() {
+    delete storage_;
```

Comment:
Have you tried to use this code without deleting and creating new storage? `config_` is passed through the pointer and it is used mainly within the `Open()` call.

Reply:
> Have you tried to use this code without deleting and creating new storage? `config_` is passed through the pointer and it is used mainly within the `Open()` call.

It seems that there is no interface to directly modify the config. Will it cause a memory leak if I don't delete it?

Reply:
I was thinking about removing the line `storage_ = new Engine::Storage(config_);` Does it work?

Reply:
And of course without deleting existing config instance.

Reply:
`config_` is a protected member so you can modify its members from the derived class (if I'm not mistaken)

Reply:
Let me try it.

Reply:
Is this your idea? I didn't delve into it, but it crashes.
```
  explicit RedisDiskTest() : TestBase() {
    // delete storage_;
    storage_->debugconfig();
    config_->RocksDB.compression = rocksdb::CompressionType::kNoCompression;
    config_->RocksDB.write_buffer_size = 1;
    // storage_ = new Engine::Storage(config_);
    Status s = storage_->Open();
    if (!s.IsOK()) {
      std::cout << "Failed to open the storage, encounter error: " << s.Msg() << std::endl;
      assert(s.IsOK());
    }
    storage_->debugconfig();
  }
```

Reply:
Yes, you are right, that was my idea. If it crashes, discard my idea :)

Reply:
Now I see. The `TestBase` constructor calls `storage_->Open()`

Reply:
> 

Yes. My guess is that there are some resources that must be destructed to be freed, so it can't just be reopened.

Reply:
> Now I see. The TestBase constructor calls storage_->Open()

Yes. My guess is that there are some resources that must be destructed to be freed, so it can't just be reopened.